tools/autograd/derivatives.yaml

```diff
@@ -161,7 +161,8 @@
   self: other.cross(grad, dim)
   other: grad.cross(self, dim)
 
-- name: cumprod  # complicated
+- name: cumprod(Tensor self, int64_t dim)
+  self: not_implemented("cumprod")  # implemented in Python
```

Comment:
This is confusing. Why do you add it here if it's going through Python anyway?

Reply:
Two reasons:

1) It's useful for people using the C++ interface
2) Once we merge Variable and Tensor, Python-implemented autograd functions will operate on Variables in the forward() (but with gradients disabled). The C++ cumprod needs to be exposed so that the Python-implemented autograd function can call it.

Reply:
I understand the first point, but I can't see how the second would work. We would need to use the C++ function in no grad mode in Python, and Python implementation in all other cases.

Reply:
I was thinking it might look something like:

```python
class Cumprod(Function):
    @staticmethod
    def forward(ctx, input, dim):
        ctx.dim = dim
        ctx.save_for_backward(input)
        return torch._C._VariableBase.cumprod(input, ctx.dim)
  
    @staticmethod
    def backward(ctx, grad_output):
        ...
```